hicommonwealth/commonwealth
# Mobile navigation pt.1

## Link to Issue
Closes: #6597 

## Description of Changes
Initial idea was to prepare scaffold first and then in another PR populate scaffold with proper components. It turned out though that I had to do a lot of style tweaks in major layout components, thus, introducing feature flag would be very overwhelming. As a result, decided to prepare the scaffold and working version of :
- left sidebar
- bottom navigation

However, for top navigation, on mobile, we won't be able to see:
- circle avatar with right-hand-side user profile drawer
- magnify glass with search view

These two things will be covered in the followup ticket #6389

Also, worth to notice, that there are things to be done in upcoming PR as well, but because of the time constraint, I decide to push first PR quicker, and take care of everything else in second PR. Thins such as:
- refactor directory structure for layout/composition components
- fix breadcrubms to look better on mobile devices
-  investigate breakpoints we use and try to unify usage of them (we should use `isWindowExtraSmall` for mobile)

## Test Plan
- use chrome devtools mobile mode or your physical device (eg with ngrok or using your local network)
- top navbar 
  - for scoped page should expand sidebar
  - for unscoped page should expand quick switcher
  - for not logged in users - sign in button should be visible
  - for logged in users - user dropdown component the same as on web (will be changed in next PR)
- bottom navigation
  - 4 icons for signed in users and 3 (no notifications) for not signed in users
  - create button should expand bottom drawer
  - other buttons should redirect to proper pages
 

https://github.com/hicommonwealth/commonwealth/assets/14819225/00143a29-03d6-4bd8-bb50-21793e1b5a61


## Deployment Plan

1. safe to deploy 

## Other Considerations

- #6389
